core/src/androidMain/kotlin/Peripheral.kt

```diff
@@ -123,6 +123,24 @@ public fun CoroutineScope.peripheral(
     )
 }
 
+public fun CoroutineScope.peripheral(
+    macAddress: String,
+    builderAction: PeripheralBuilderAction = {},
+): Peripheral {
+    val builder = PeripheralBuilder()
+    builder.builderAction()
+    val bluetoothDevice = BluetoothAdapter.getDefaultAdapter().getRemoteDevice(macAddress)
```

Comment:
Good point.  I can implement this with null unless there is consensus for an Exception in stead.  Essentially it would work more similarly to the IOS implementation where there is a search for existing devices with same id and return null if none exist.

Reply:
For others following this thread, I responded via https://github.com/JuulLabs/kable/pull/280#discussion_r832888616.